### PR TITLE
Static test fixes

### PR DESCRIPTION
**Pull request by matmair**
See #2989
Fixes #2942

<a href="https://gitpod.io/#https://github.com/inventree/InvenTree/pull/2991"></a>

### CODE REVIEW

**Comment by matmair** [comment]
@SchrodingersGat any idea what is the difference between sqlite and postgres/mysql tests? I can't get my head around why static data is not accessible even though it is clearly collected on startup.

**Comment by SchrodingersGat** [comment]
@matmair not sure, I have found in the past that PK values are not auto-assigned in a consistent manner when running postgres  mysql tests. They seem to be repeatable for sqlilte. Maybe it is something to do with that?

**Comment by matmair** [comment]
@SchrodingersGat everything is working except on sqlite sometimes. I included a check for that. The coverage is much better this way though so I would suggest merging if you agree.

File: InvenTree/InvenTree/test_middleware.py [hunk 1]
```diff
@@ -32,6 +32,10 @@ def test_AuthRequiredMiddleware(self):
         # logout
         self.client.logout()
 
+        # check that static files go through
+        # TODO @matmair reenable this check
+        # self.check_path('/static/css/inventree.css', 302)
```
Comment (SchrodingersGat) [inline InvenTree/InvenTree/test_middleware.py:37]: Does this still need to be re-enabled for this PR?

File: InvenTree/label/apps.py [hunk 2]
```diff
@@ -42,6 +43,16 @@ def create_labels(self):
         """
         Create all default templates
         """
+        # Test if models are ready
+        try:
+            from .models import StockLocationLabel
+            obj = StockLocationLabel.objects.first()
+            print(obj)
```
Comment (SchrodingersGat) [inline InvenTree/label/apps.py:50]: Remove this debug?

**Review by SchrodingersGat — APPROVED** [review]

File: InvenTree/InvenTree/test_middleware.py (hunk 1 above)
Comment (matmair) [reply InvenTree/InvenTree/test_middleware.py:37]: @schrodingergat I would like to leave it in so we don't forget to fix this behaviour. Todos with my name attached stick out in my CI so I am reminded about them regularly.

File: InvenTree/label/apps.py (hunk 2 above)
Comment (matmair) [reply InvenTree/label/apps.py:50]: Absolutely!

**Comment by matmair** [comment]
@SchrodingersGat requested changes are in.

**Review by SchrodingersGat — APPROVED** [review]

**Comment by SchrodingersGat** [comment]
Nice work. I might have to employ some of these changes to get https://github.com/inventree/InvenTree/pull/3037 to work

**Pull request merged**